src/app/components/wrapper-components/online-game-wrapper/online-game-wrapper.component.ts

```diff
@@ -297,20 +297,23 @@ export class OnlineGameWrapperComponent extends GameWrapper<MinimalUser> impleme
         Utils.assert(success.isSuccess(), 'Chosen move should be legal after all checks, but it is not! Reason: ' + success.getReasonOr(''));
         this.gameComponent.node = success.get();
         if (this.role === PlayerOrNone.NONE) {
-            await this.updateBoardAndShowLastMove(isLastMoveOfBatch);
+            await this.showNextMove(isLastMoveOfBatch);
         } else {
             // We only animate the move of opponent, because the users move has already been animated before sending it
             const triggerAnimation: boolean = currentPartTurn % 2 !== this.role.getValue();
-            await this.updateBoardAndShowLastMove(triggerAnimation && isLastMoveOfBatch);
+            await this.showNextMove(triggerAnimation && isLastMoveOfBatch);
         }
-        this.setCurrentPlayerAccordingToCurrentTurn();
+        await this.setCurrentPlayerAccordingToCurrentTurn();
         this.timeManager.onReceivedMove(moveEvent);
         this.requestManager.onReceivedMove();
     }
 
-    private setCurrentPlayerAccordingToCurrentTurn(): void {
+    private async setCurrentPlayerAccordingToCurrentTurn(): Promise<void> {
         this.currentUser = this.players[this.gameComponent.getTurn() % 2].get();
-        this.gameComponent.setInteractive(this.currentUser.name === this.getPlayer().name);
+        await this.setInteractive(
+            this.currentUser.name === this.getPlayer().name,
+            false,
```

Comment:
WOWwat, setInteractive avec deux paramètres, mais quel diabolesserie esse ?

Reply:
![image](https://github.com/EveryBoard/EveryBoard/assets/41200522/796f73ab-f603-4767-98df-b72b23491d14)

Reply:
on à combien d'appels à setInteractive ? Est-ce que ces appels pourraient pas eux appeler updateBoard quand c'est nécessaire ? Car là c'est louche : setInteractive fait "setInteractive" ET "updateBoard", ça facilite pas la compréhension du flow parmi tous ces composants

Reply:
DCWC: une seule fois à true... c'qui semble bizarre ?
TGWC: dans showStep on le set à false sauf si c'est une information.
LGWC: dans updatePlayer on le set et dans proposeAIToPlay
OGWC: onGameEnd et dans setCurrentPlayerAccordingToCurrentTurn

Sinon quand on set l'interactivité, faut bien dire au composant enfant de se redessiner, il a des marqueurs à rajouter ?
J'propose qu'on liste içi les évènements qui triggeurent le lifecycle d'un composant, et qu'après celui-ci la prochaine itération double check dans chaque scénario qu'y a pas de double appels où d'appels non nécessaire (c'qui dépend du jeu, si c'est legit sur un jeu, soit on le change pour qu'il s'en passe, soit c'est c'est vraiment légit et bah, eh, on garde l'appel)

Reply:
yep bon argument le coup des marqueurs (même si du coup c'est pas vraiment "updateBoard", mais bon).
Du coup bah ça me semble bon ainsi non, on peut résoudre ?